### PR TITLE
ISLANDORA-1535

### PR DESCRIPTION
**Pull request by ajs6f**

### CODE REVIEW

**Comment by manez** [comment]
Hi @ajs6f  - glad to see a new name show up offering a commit! We cover all contributions to the Islandora stack with Contributor License Agreements, if you wouldn't mind filling one out and sending it my way at manez@islandora.ca 

http://islandora.ca/sites/default/files/islandora_cla.pdf

**Comment by ajs6f** [comment]
This was/is work-for-hire, so let me clear this with my principals (I have no reason to think there will be any problem).

**Comment by ajs6f** [comment]
@manez On its way.

**Comment by jordandukart** [comment]
@ruebot,

@willtp87 will be away until Monday November 16th, he won't be available to have a look at this until at least then.

## Thanks :smile:!

Jordan Dukart
_Developer_
_[discoverygarden inc.](http://www.discoverygarden.ca) | Managing Digital Content_

**Comment by DiegoPino** [comment]
Thanks @jordandukart :smile: . And welcome @ajs6f !

**Comment by DiegoPino** [comment]
@ajs6f, hi, i just did an first independent review of your pull and it looks fine. There are some minor coding standards issues, like not documenting your new sparql endpoint argument `$sparqlEndpoint` using `@param` and maybe some control structures that i'm sure someone will comment on (not me!)

Also, would be nice if we could get the islandora tuque wrapper interface to enable this option from islandora itself/drupal, if this pull gets though committers approval, maybe a new ticket for that.
https://github.com/Islandora/islandora/search?utf8=✓&q=+RepositoryConnection

Lets wait for @willtp87 review, probably will need some integration tests(human ones, not the fine automated ones you already did!) using an external tripli and some documentation also on this.

Very, very useful and interesting pull (smile).

Lastly, how does it wok on your side?

**Comment by ajs6f** [comment]
I'll fix that @param oversight in a commit later today, and I will check into the coding standards and see if I can make other corrections on my own. I was hoping to get on to exactly the kind of changes for Islandora proper you describe above, but I was not sure where to begin-- your comment helps me greatly with that!
We haven't tried this out yet, because we are still working on the mechanics of migrating RDF stores, which has nothing to do with Islandora. But our timelines are such that I need to pursue both lines of work in parallel.

File: RepositoryConnection.php [hunk 1]
```diff
@@ -45,13 +46,21 @@ class RepositoryConnection extends CurlConnection implements RepositoryConfigInt
    *   The username to connect with.
    * @param string $password
    *   The password to connect with.
+   * @param string $sparqlEndpoint
+   *   The URL of a SPARQL Query Endpoint, defaults to $url.
    */
-  function __construct($url = self::FEDORA_URL, $username = NULL, $password = NULL) {
+  function __construct($url = self::FEDORA_URL, $username = NULL, $password = NULL, $sparqlEndpoint = "DUMMY") {
```
Comment (nigelgbanks) [inline RepositoryConnection.php]: Minor style note, perhaps using NULL, instead of a placeholder string?

File: tests/scripts/travis_setup.sh [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
 
-# Java 8, if needed.
-if [ $FEDORA_VERSION = "3.8.1" ]; then
-  sudo add-apt-repository -y ppa:webupd8team/java
-  sudo apt-get update
-  sudo apt-get install -y oracle-java8-installer oracle-java8-set-default
-  sudo update-java-alternatives -s java-8-oracle
-  export JAVA_HOME=/usr/lib/jvm/java-8-oracle
-fi
+# Java 8
```
Comment (nigelgbanks) [inline tests/scripts/travis_setup.sh:3]: Will Java 8 work in older versions of Fedora from the .travis.yaml?

File: RepositoryConnection.php (hunk 1 above)
Comment (ajs6f) [inline RepositoryConnection.php]: Quite right, @nigelgbanks. I will correct that.

File: tests/scripts/travis_setup.sh (hunk 2 above)
Comment (ajs6f) [inline tests/scripts/travis_setup.sh:3]: It certainly should. It did for me, in Travis. That having been said, we can refine this to use a matrix formulation for Java version as we do for Fedora version.

**Comment by ajs6f** [comment]
@ruebot I see [here](https://github.com/Islandora/tuque/blob/1.x/CONTRIBUTING.md) that Tuque asks for [Drupal Coding Standards](https://www.drupal.org/coding-standards). Is there an automatic way to check that?

**Comment by ajs6f** [comment]
@ruebot ++ 
My nostalgic PHP adventure continues...

**Comment by ajs6f** [comment]
@ruebot Looks like there is [help](https://www.drupal.org/node/1420004) for us Eclipse people, too.

**Comment by ajs6f** [comment]
@ruebot In Eclipse we tend to just have these types of validations on all the time, so you get instant feedback if you type some ill-formed code. And I can now report that Tuque is definitely not straight with Drupal coding standards (~470 violations). Whether that matters at all is well beyond my competence, of course!

**Comment by ajs6f** [comment]
@ruebot That's one way to do. The other, of course, is to do the boring nitpicky diffs to bring it up to spec. I guess whether that's worthwhile depends on whether you expect to get many more contributions to this module.

**Comment by qadan** [comment]
_pokes head in because automated testing_ We could actually bring in Travis to run coder on this with Drupal standards! We just don't, currently. I feel like perhaps we should, even if it doesn't need much more updating? Tuque is obfuscated enough as it is without having to worry about whether its innards will be properly documented or readable. The wiki docs can't tell you _everything_ you could ever need to know.

**Comment by ajs6f** [comment]
I didn't mean to get off onto a tangent here. {grin} Maybe I should open another issue into which to direct conversation about Tuque's code standards?

**Comment by ajs6f** [comment]
@ruebot https://jira.duraspace.org/browse/ISLANDORA-1546

**Comment by qadan** [comment]
Thanks! It's my dream to have all this stuff cleaned up and in order haha :D

**Comment by willtp87** [comment]
@ruebot 

As per discussion in [ISLANDORA-1535](https://jira.duraspace.org/browse/ISLANDORA-1535) a way forward was identified for the ticket regarding this pull request and [the other](https://github.com/Islandora/islandora/pull/636) in late January.
I am uncertain as to if @ajs6f  will be requiring these pulls going forward.

**Pull request closed without merging**